### ISSUE TITLE
Use of unknown class

### ISSUE THREAD

**Issue by ksn135** [DEVSENSE/phptools-docs#284]
Greetings gentlemen! Great product! 
However, it is VERY annoying to get bogus messages about unknown classes, especially if they are declared correctly above.
Maybe I'm doing something wrong?
```v1.30.12518```
<img width="1494" alt="Screenshot 2023-02-15 at 11 58 12" src="https://user-images.githubusercontent.com/230304/218980888-7aefc948-b6da-435e-879d-76bb3b27055d.png">
<img width="2167" alt="Screenshot 2023-02-15 at 11 58 46" src="https://user-images.githubusercontent.com/230304/218980982-424a9ff6-25ef-4645-a0f1-9eb5c23ab6dc.png">

**Comment by jakubmisek** [comment]
Hello Serg! and thank you for reporting the issue.

I'm glad you like our tools.

The class `AbstractDashboardController` was probably not parsed - is it included in your workspace? The `use` itself doesn't matter here.
The same for the other classes.

**Comment by ksn135** [comment]
> The class `AbstractDashboardController` was probably not parsed - is the file with the class included in your workspace?

No, it's in vendor directory

**Comment by jakubmisek** [comment]
> No, it's in vendor directory

Thanks; and is the `vendor` in your workspace? or is it excluded? We are probably not processing the `vendor` directory if it's not listed in VSCode Explorer, but it will be easy to fix.

**Comment by jakubmisek** [comment]
just check please if it's there

![image](https://user-images.githubusercontent.com/842150/219061369-1a439f48-5ab9-471a-869e-4000ca9769d5.png)

**Comment by ksn135** [comment]
No (((
<img width="617" alt="Screenshot 2023-02-15 at 18 08 55" src="https://user-images.githubusercontent.com/230304/219066815-eda603f4-6599-4570-a077-234b07f85a74.png">

My settings.json:
```json
{
   "php.executables": {
        "v8.1": "/usr/local/Cellar/php/8.1.12/bin/php"
    },
    "editor.unicodeHighlight.allowedLocales": {
        "ru": true
    },
    "php-cs-fixer.executablePath": "/usr/local/bin/php-cs-fixer",
    "php-cs-fixer.lastDownload": 1669389209649,
    "php-cs-fixer.exclude": [],
    "php-cs-fixer.formatHtml": true,
    "php-cs-fixer.onsave": true,
    "php.problems.excludeGitIgnore": false,
    "php.problems.exclude": {
        "/var/": true,
        "/vendor/**": true,
        ".php-cs-fixer.dist.php": true
    },
    "php.version": "v8.1",
    "php-cs-fixer.autoFixBySemicolon": true,
    "editor.multiCursorModifier": "ctrlCmd",
    "phpGetterSetter.short": true,
    "php.format.rules.alignConsecutiveAssignments": false,
    "php.format.rules.callParametersNewLineBeforeRightParen": false,
    "php.format.rules.spaceWithinDeclParens": false,
    "editor.formatOnSave": true,
    "[jsonc]": {
        "editor.defaultFormatter": "vscode.json-language-features"
    },
    "php.hover.fullname": true,
    "php.inlayHints.parameters.suppressNameMatchingValue": true,
    "php.linkedEditing.variables": true,
    "php.completion.autoimport-docblock": "auto-import",
    "php.inlayHints.parameters.enabled": false,
}
```

**Comment by ksn135** [comment]
Well, remove vendor dir from php.problems.exclude.
That didn't help. AbstractDashboardController is still unknown.
BUT I have dot at vendor now and 5K+ errors ))
<img width="1370" alt="Screenshot 2023-02-15 at 18 24 18" src="https://user-images.githubusercontent.com/230304/219072547-ea068b0e-3614-4af2-a50b-89e06e88cec5.png">
<img width="414" alt="Screenshot 2023-02-15 at 18 28 44" src="https://user-images.githubusercontent.com/230304/219072832-fa9078eb-1879-443d-a5dd-53b23fdaabcb.png">

**Comment by ksn135** [comment]
Switched to Release version (v1.30.12484). 
That didn't help either. 
Something broken ((
Current config:
```json
{
    "php.format.codeStyle": "PSR-12",
    "php.problems.excludeGitIgnore": false,
    "php.problems.scope": "opened",
    "php.version": "v8.1",
    "php.format.rules.alignConsecutiveAssignments": false,
    "php.format.rules.callParametersNewLineBeforeRightParen": false,
    "php.format.rules.spaceWithinDeclParens": false,
    "php.hover.fullname": true,
    "php.inlayHints.parameters.suppressNameMatchingValue": true,
    "php.linkedEditing.variables": true,
    "php.completion.autoimport-docblock": "auto-import",
    "php.inlayHints.parameters.enabled": false,
}
```
Vendor directory WITHOUT "dot":
<img width="587" alt="Screenshot 2023-02-15 at 18 44 00" src="https://user-images.githubusercontent.com/230304/219077320-4bcbdd25-6931-4871-9426-7d9c3400ea15.png">
Visual Studio Code:
```
Version: 1.75.1
Commit: 441438abd1ac652551dbe4d408dfcec8a499b8bf
Date: 2023-02-08T21:34:59.000Z
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Darwin x64 22.2.0
Sandboxed: No
```

**Comment by jakubmisek** [comment]
Not sure what the dot means :)

Just please check the file with `class AbstractDashboardController` is in your workspace.

"Unknown class" happens, when the class declaration is not found in your workspace. Search for the file `AbstractDashboardController.php`.

**Comment by ksn135** [comment]
Thank you so much for your help! 
I still think there was a settings conflict. I did two things. First, I created a new workspace and added a project folder there, as well as two additional separate folders with a copy of the bundles that are in the vendors. Second, I carefully went through the settings.json and removed everything with the php prefix except the necessary minimum.
So far, everything works as it was originally supposed to.
Thanks again for a great product!

```json
    "php.executables": {
        "v8.1": "/usr/local/Cellar/php/8.1.12/bin/php"
    },
    "php.version": "v8.1",
    "php.format.codeStyle": "Off",
```

<img width="613" alt="Screenshot 2023-02-16 at 03 34 53" src="https://user-images.githubusercontent.com/230304/219227781-3e5d475f-f222-41c7-8a8f-150186643228.png">

<img width="560" alt="Screenshot 2023-02-16 at 03 35 49" src="https://user-images.githubusercontent.com/230304/219227874-dbfdb954-a629-4607-a196-891d65eb1d09.png">

<img width="557" alt="Screenshot 2023-02-16 at 03 37 20" src="https://user-images.githubusercontent.com/230304/219227912-66de2ee1-7870-4b66-a1d3-3ea969027816.png">

PS: By the way, it just hit me, maybe all the problems are due to symlinks? If the vendors folder has symbolic links, then your engine is looking for classes in them? Maybe that's the problem? 

<img width="778" alt="Screenshot 2023-02-16 at 03 42 27" src="https://user-images.githubusercontent.com/230304/219228608-3e1599e0-93cf-4f74-8679-86bd9488f4fd.png">
